starter-code/src/clue.js

```diff
@@ -92,3 +92,158 @@ var roomsArray = [];
+  first_name: "Victor",
+  last_name: "Plum",
+  color: "purple",
+  description: "Billionare video game designer",
+  age: 22,
+  image: "https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg",
+  occupation: "Designer"
+};
+
+var missScarlet = {
+  first_name: "Kasandra",
+  last_name: "Scarlet",
+  color: "red",
+  description: "She is an A-list movie star with a dark past",
+  age: 31,
+  image: "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+  occupation: "Actor"
+};
+
+var mrsPeacock = {
+  first_name: "Eleanor",
+  last_name: "Peacock",
+  color: "blue",
+  description:
+    "She is from a wealthy family and uses her status and money to earn popularity",
+  age: 36,
+  image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+  occupation: "Socialité"
+};
+
+var mrMustard = {
+  first_name: "Jack",
+  last_name: "Mustard",
+  color: "yellow",
+  description:
+    "He is a former football player who tries to get by on his former glory",
+  age: 62,
+  image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+  occupation: "Retired Football player"
+};
+
+charactersArray.push(
+  mrGreen,
+  drOrchid,
+  profPlum,
+  missScarlet,
+  mrsPeacock,
+  mrMustard
+);
+
+var rope = { name: "rope", weight: 10 };
+var knife = { name: "knife", weight: 8 };
+var candlestick = { name: "candlestick", weight: 2 };
+var dumbbell = { name: "dumbbell", weight: 30 };
+var poison = { name: "poison", weight: 2 };
+var axe = { name: "axe", weight: 15 };
+var bat = { name: "bat", weight: 13 };
+var trophy = { name: "trophy", weight: 25 };
+var pistol = { name: "pistol", weight: 20 };
+
+weaponsArray.push(
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+);
+
+var dinningRoom = { name: "Dinning Room" };
+var conservatory = { name: "Conservatory" };
+var kitchen = { name: "Kitchen" };
+var study = { name: "Study" };
+var library = { name: "Library" };
+var billiardRoom = { name: "Billiard Room" };
+var lounge = { name: "Lounge" };
+var ballroom = { name: "Ball Room" };
+var hall = { name: "Hall" };
+var spa = { name: "Spa" };
+var livingRoom = { name: "Living Room" };
+var observatory = { name: "Observatory" };
+var theater = { name: "Theater" };
+var guestHouse = { name: "Guest House" };
+var patio = { name: "Patio" };
+
+roomsArray.push(
+  dinningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+);
+
+var randomSelector = function(array) {
+  return Math.floor(Math.random() * array.length);
+};
+
+var deck = [charactersArray, weaponsArray, roomsArray];
+var misteryEnvelope = [];
+
+for (i = 0; i < deck.length; i++) {
+  var pick = randomSelector(deck[i]);
+  mysteryEnvelope.push(deck[i][pick]);
+}
+
+console.log(mysteryEnvelope);
+
+function revealMistery() {
+  console.log(
+    mysteryEnvelope[0].first_name +
```

Comment:
pay attention to your spelling- you've wrote both  `mysteryEnvelope` and `misteryEnvelope` within the same function. Your initial array is spelt `misteryEnvelope` 

You also need to pass misteryEnvelope into your function